src/StatusQ/Controls/StatusTagSelector.qml

```diff
@@ -0,0 +1,121 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+//TODO check colors, sizes, margins, general clean up
+Item {
+    id: root
+
+    implicitWidth: 448
+    implicitHeight: 44
+
+    property alias textEdit: edit
+    property alias text: edit.text
+    property string warningText: ""
+    property int nameCountLimit: 5
+    property ListModel namesModel: ListModel { }
+
+    function find(model, criteria) {
+      for (var i = 0; i < model.count; ++i) if (criteria(model.get(i))) return model.get(i);
+      return null;
+    }
+
+    function insertTag(name, id) {
+        if (!find(namesModel, function(item) { return item.publicId === id }) && namesModel.count < root.nameCountLimit) {
+            namesModel.insert(namesModel.count, {"name": name, "publicId": id});
+            edit.clear();
+        }
+    }
+
+    Rectangle {
+        anchors.fill: parent
+        radius: 8
+        color: Theme.palette.baseColor2
+        border.width: 1
+        border.color: edit.activeFocus ? Theme.palette.primaryColor2 : "transparent"
+
+        RowLayout {
+            anchors.fill: parent
+            anchors.leftMargin: 16
+            anchors.rightMargin: 16
+            spacing: 8
+            StatusBaseText {
+                Layout.preferredWidth: 22
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignLeft
+                color: Theme.palette.baseColor1
+                text: qsTr("To: ")
+            }
+
+            ListView {
+                id: namesList
+                //TODO check if 220 is the correct limit for width
+                Layout.preferredWidth: (childrenRect.width > 220) ? 220 : childrenRect.width
+                implicitHeight: 30
+                visible: (count > 0)
+                Layout.alignment: Qt.AlignVCenter
+                model: namesModel
+                orientation: ListView.Horizontal
+                spacing: 8
+                clip: true
+                onContentWidthChanged: {
+                    positionViewAtEnd();
+                }
+
+                delegate: Rectangle {
+                    id: nameDelegate
+                    width: (nameText.contentWidth + 30)
+                    height: 30
+                    color: Theme.palette.primaryColor1
+                    radius: 8
+                    StatusBaseText {
+                        id: nameText
+                        anchors.left: parent.left
+                        anchors.leftMargin: 8
+                        anchors.verticalCenter: parent.verticalCenter
+                        color: Theme.palette.white
+                        text: name
+                    }
+                    ToolButton {
+                        width: 24
+                        height: 24
```

Comment:
Not exactly sure what `ToolButton` is, but the size seems to be off. The icon needs to be rendered a bit bigger.

Also, can you ensure this turns into a cursor pointer when hovered?

Reply:
part of WIP, changed to `StatusIconTabButton`